File: src/main/java/com/reelcosmos/service/auth/UserService.java
package com.reelcosmos.service.auth;

import com.reelcosmos.dto.request.UserUpdateRequest;
import com.reelcosmos.dto.response.UserResponse;
import com.reelcosmos.entity.Role;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import java.util.List;

public interface UserService {


    UserResponse getCurrentUser();


    UserResponse getUserById(Long id);

    Page<UserResponse> getUsers(
            String search,
            Pageable pageable
    );

    UserResponse updateCurrentUser(
            UserUpdateRequest request
    );


    void deleteCurrentUser();



    // ================= ADMIN =================


    UserResponse updateUser(
            Long id,
            UserUpdateRequest request
    );


    void deleteUser(
            Long id
    );


    UserResponse changeUserRole(
            Long id,
            Role role
    );

}

File: frontend/src/services/userService.ts
import api from "../api/axios";


import type {
    User,
    UserUpdateRequest,
} from "../types/user";


const USERS_URL = "/api/users";



export const getCurrentUser =
    async (): Promise<User> => {


        const response =
            await api.get<User>(
                `${USERS_URL}/me`
            );


        return response.data;

    };



export const updateCurrentUser =
    async (
        data: UserUpdateRequest
    ): Promise<User> => {


        const response =
            await api.put<User>(
                `${USERS_URL}/me`,
                data
            );


        return response.data;

    };



export const deleteCurrentUser =
    async (): Promise<void> => {


        await api.delete(
            `${USERS_URL}/me`
        );

    };